src/devices/Lps22hb/Lps22hb.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Diagnostics;
+using UnitsNet;
+
+namespace Iot.Device.Lps22hb
+{
+    /// <summary>
+    /// LPS22HB - Piezoresistive pressure and thermometer sensor
+    /// </summary>
+    [Interface("LPS22HB - Piezoresistive pressure and thermometer sensor")]
+    public class Lps22hb : IDisposable
+    {
+        private const byte ReadMask = 0x80;
+        private const byte Lps22hID = 0xB1;
+        private I2cDevice _i2c;
+
+        /// <summary>
+        /// Default I2C address
+        /// </summary>
+        public const int DefaultI2cAddress = 0x5C;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Lps22hb"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Lps22hb(I2cDevice i2cDevice)
+        {
+            if (i2cDevice == null)
+            {
+                throw new ArgumentNullException(nameof(i2cDevice));
+            }
+
+            _i2c = i2cDevice;
+
+            if (!TryRead())
+            {
+                throw new Exception("Device not found");
```

Comment:
Do not throw this generic exception type. Use IOException or similar. In this case, the exception can only happen if there is a device on the provided address but not the expected one, as i2cDevice.Read() in itself throws an exception if there's no answer from the bus.